mmv1/third_party/terraform/tests/data_source_google_compute_snapshot_test.go

```diff
@@ -0,0 +1,205 @@
+package google
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccSnapshotDatasource_name(t *testing.T) {
+	t.Parallel()
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSnapshot_name(getTestProjectFromEnv(), randString(t, 10)),
+				Check: resource.ComposeTestCheckFunc(
+					checkDataSourceStateMatchesResourceStateWithIgnores(
+						"data.google_compute_snapshot.default",
+						"google_compute_snapshot.default",
+						map[string]struct{}{"zone": {}},
+					),
+				),
+			},
+		},
+	})
+}
+
+func TestAccSnapshotDatasource_filter(t *testing.T) {
+	t.Parallel()
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSnapshot_filter(getTestProjectFromEnv(), randString(t, 10)),
+				Check: resource.ComposeTestCheckFunc(
+					checkDataSourceStateMatchesResourceStateWithIgnores(
+						"data.google_compute_snapshot.default",
+						"google_compute_snapshot.c",
+						map[string]struct{}{"zone": {}},
+					),
+				),
+			},
+		},
+	})
+}
+
+func TestAccSnapshotDatasource_filter_mostRecent(t *testing.T) {
```

Comment:
```suggestion
func TestAccSnapshotDatasource_filterMostRecent(t *testing.T) {
```

nit: We generally use a `_` between the resource/datasource and test case name by convention, but camel case the rest.

Reply:
Noted and Updated.